src/main/java/org/isf/medicalstock/service/MedicalStockIoOperations.java

```diff
@@ -96,12 +100,17 @@ public List<Integer> getMedicalsFromLot(String lotCode) throws OHServiceExceptio
 	 * @throws OHServiceException
 	 */
 	public List<Movement> newAutomaticDischargingMovement(Movement movement) throws OHServiceException {
+		List<Movement> dischargingMovements = new ArrayList<>();
 		List<Lot> lots = getLotsByMedical(movement.getMedical());
-
+		Medical medical = movement.getMedical();
 		int qty = movement.getQuantity(); // movement initial quantity
-		List<Movement> dischargingMovements = new ArrayList<>();
+
+		if (lots.isEmpty()) {
+			LOGGER.warn("No lots with available quantity found for medical {}", medical.getDescription());
+			return dischargingMovements;
+		}
 		for (Lot lot : lots) {
-			Movement splitMovement = new Movement(movement.getMedical(), movement.getType(), movement.getWard(),
+			Movement splitMovement = new Movement(medical, movement.getType(), movement.getWard(),
```

Comment:
`medical` is extracted into a local variable so it doesn't need to be fetched over and over; then why aren't `type` and `ward` done the same way?

Reply:
`type` and `ward` are extracted only once. Maybe we can do it for `lot.getCode()` inside the `for` loop?

Reply:
It is inside of a loop so for every iteration they are fetched, aren't they?

Reply:
Yes, I mean extract the lot.getCode() at the begin of the loop

Reply:
done anyway